plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TableSnapshot.java

```diff
@@ -190,75 +193,30 @@ public Stream<DeltaLakeTransactionLogEntry> getCheckpointTransactionLogEntries(
             return Stream.empty();
         }
 
-        LastCheckpoint checkpoint = lastCheckpoint.get();
         // Add entries contain statistics. When struct statistics are used the format of the Parquet file depends on the schema. It is important to use the schema at the time
         // of the Checkpoint creation, in case the schema has evolved since it was written.
         if (entryTypes.contains(ADD)) {
             checkState(metadataAndProtocol.isPresent(), "metadata and protocol information is needed to process the add log entries");
         }
 
-        Stream<DeltaLakeTransactionLogEntry> resultStream = Stream.empty();
-        for (Location checkpointPath : getCheckpointPartPaths(checkpoint)) {
-            TrinoInputFile checkpointFile = fileSystem.newInputFile(checkpointPath);
-            resultStream = Stream.concat(
-                    resultStream,
-                    stream(getCheckpointTransactionLogEntries(
-                            session,
-                            entryTypes,
-                            metadataAndProtocol.map(MetadataAndProtocolEntry::metadataEntry),
-                            metadataAndProtocol.map(MetadataAndProtocolEntry::protocolEntry),
-                            checkpointSchemaManager,
-                            typeManager,
-                            stats,
-                            checkpoint,
-                            checkpointFile,
-                            partitionConstraint,
-                            addStatsMinMaxColumnFilter)));
-        }
-        return resultStream;
```

Comment:
it would be sufficient to just replace Stream.concat with Stream.flatMap.
at least per https://stackoverflow.com/a/64122812 this should be sufficient.

Reply:
Verified through debugging and indeed it works.